packages/miniapp-utils/src/helpers/i18n.tsx

```
@@ -0,0 +1,583 @@
+        }
+    }
+
+    /**
+     * Obtains locale preference from query parameter (if specified), cookie, request.headers['accept-language'] or window.navigator.languages
+     * and then selects supported locale using selectSupportedLocale method
+     */
+    getPreferredLocale (req: Optional<IncomingMessage>, res: Optional<ServerResponse>): LocaleSelection<AvailableLocale> {
+        // Step 1: Query must be resolved before any cookies, since it's more explicit
+        if (this.localeQueryParam) {
+            let paramValue: string | null = null
+            if (req && req.url) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)